### PR TITLE
Adding playbook to test svc deletion

### LINKED ISSUE #1214: delete volume should clear associated svc
BUG REPORT

I created and deleted an applications that uses openebs storageclass. The storage pods and deployments created by the m-apiserver are deleted but not the "service" I had to manually delete them. 

```
kiran_mova@kmova-minikube:~$ kubectl --server="http://10.128.0.12:8080" get pods
No resources found.
kiran_mova@kmova-minikube:~$ kubectl --server="http://10.128.0.12:8080" get pvc
No resources found.
kiran_mova@kmova-minikube:~$ kubectl --server="http://10.128.0.12:8080" get svc
NAME                                                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
kubernetes                                          ClusterIP   10.96.0.1       <none>        443/TCP             4d
pvc-6b56c586-0b3d-11e8-b0d4-42010a80000c-ctrl-svc   ClusterIP   10.108.53.3     <none>        3260/TCP,9501/TCP   6h
pvc-b6b83c42-0b1f-11e8-b0d4-42010a80000c-ctrl-svc   ClusterIP   10.97.212.206   <none>        3260/TCP,9501/TCP   10h
kiran_mova@kmova-minikube:~$ kubectl --server="http://10.128.0.12:8080" delete svc pvc-6b56c586-0b3d-11e8-b0d4-42010a80000c-ctrl-svc
service "pvc-6b56c586-0b3d-11e8-b0d4-42010a80000c-ctrl-svc" deleted
kiran_mova@kmova-minikube:~$ kubectl --server="http://10.128.0.12:8080" delete svc pvc-b6b83c42-0b1f-11e8-b0d4-42010a80000c-ctrl-svc
service "pvc-b6b83c42-0b1f-11e8-b0d4-42010a80000c-ctrl-svc" deleted
kiran_mova@kmova-minikube:~$ 
```

### LINKED ISSUE #1237: CI test for volume delete when replica count > no. of K8s nodes
It has been seen that volume service is not deleted for a few edge cases.
e.g. when replica count < no of k8s nodes available.

This issue will track the implementation of ci test case for above.

This issue will verifiy: https://github.com/openebs/openebs/issues/1214

### PR DESCRIPTION
**Pull request by gprasath**
**What this PR does / why we need it**:
With this PR, adding a playbook which checks for successful deletion of SVCs and pods during PVC deletion.
 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 
fixes #1237 
verifies # https://github.com/openebs/maya/pull/263

**Special notes for your reviewer**:
- This playbook does the below actions.
- Check the number of nodes in K8s cluster.
- Replace replica count in storage class yaml by providing the count greater than the number of nodes.
- Create storage class and replica.
- Check  if it gets created.
- Delete PVC and check if the pods and svc are deleted successfully.

### CODE REVIEW

File: e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      
+       - name: Check if the svc is created
+         shell: kubectl get svc | grep pvc | grep ctrl
+         args:
+           executable: /bin/bash
+         register: svc
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}" 
+         until: "'<none>' in svc.stdout"
+         delay: 100
+         retries: 3
```
Comment (ksatchit) [inline e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml]: - Let's check against the PV name which can be derived after deploying create_pvc.yaml (there may be residues from previous tests)
- The service gets created almost instantaneously after deploying. Better would be carry out this check after pod creation is successful, i.e., replica and controller rollout is complete (there is ideally no need to have a 100s delay b/w retries)

File: e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+         until: "'<none>' in svc.stdout"
+         delay: 100
+         retries: 3      
+
+       - name: Getting service name
+         shell: kubectl get svc | grep svc | awk {'print $1'}
+         args:
+           executable: /bin/bash
+         register: svc_name
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
```
Comment (ksatchit) [inline e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml]: Again, let us use the PV name by which to extract the corresponding svc name

File: e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+           executable: /bin/bash
+         register: result
+         until: result.stdout|int >= 1
+         delay: 30
+         retries: 10
+         with_items:
+           - ctrl
+           - rep
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
```
Comment (ksatchit) [inline e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml]: With no of replicas > node count (by 1 in this test), one of the replicas should be unscheduled. Let us track that as well as part of the behaviour verification.
